### PR TITLE
Plugins: Add forward_host_env_vars setting

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Similar to https://github.com/grafana/grafana/pull/79296, moves the `forward_host_env_vars` plugin settings under `[plugins]`, as a comma-separated list of plugin IDs:

```ini
[feature_toggles]
pluginsSkipHostEnvVars = true

[plugin.SOME_ID]
forward_host_env_vars = true

[plugin.OTHER_ID]
forward_host_env_vars = true
```

:point_down: 

```ini
[feature_toggles]
pluginsSkipHostEnvVars = true

[plugins]
forward_host_env_vars = SOME_ID, OTHER_ID
```

This also allows to configure the setting via the env var `GF_PLUGINS_FORWARD_HOST_ENV_VARS_PLUGINS`

**Why do we need this feature?**

Consistency and allows using environment variables to configure the setting.

**Who is this feature for?**

Grafana users.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by wbrowne — APPROVED** [review]
Nice one! LGTM

File: pkg/setting/setting_plugins.go [hunk 1]
```diff
@@ -78,3 +55,18 @@ func (cfg *Cfg) readPluginSettings(iniFile *ini.File) error {
 
 	return nil
 }
+
+// readPluginIDsList takes a config value containing a list of plugin ids separated by commas and returns the
+// corresponding slice of strings. Each value has its leading and trailing whitespace removed. Empty values are
+// discarded.
+func readPluginIDsList(configValue string) []string {
```
Comment (marefr) [inline pkg/setting/setting_plugins.go:62]: FYI the below util function is used throughout the setting package so we should maybe reuse that for consistency?

https://github.com/grafana/grafana/blob/106903b549d8744597144cc5151fd0b08b210973/pkg/util/strings.go#L32-L48

**Review by marefr — APPROVED** [review]
LGTM with a comment

File: pkg/setting/setting_plugins.go (hunk 1 above)
Comment (xnyo) [reply pkg/setting/setting_plugins.go]: Thanks, I was not aware of this! I have updated it so it uses `util.SplitString` instead

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**